SingularityService/src/main/java/com/hubspot/singularity/data/CuratorAsyncManager.java

```diff
@@ -40,12 +41,22 @@ public CuratorAsyncManager(CuratorFramework curator, SingularityConfiguration co
       paths.add(ZKPaths.makePath(parent, child));
     }
 
-    return getAsyncThrows(parent, paths, transcoder);
+    return getAsyncThrows(parent, paths, transcoder, Optional.<ZkCache<T>> absent());
   }
 
-  private <T> List<T> getAsyncThrows(final String pathNameForLogs, final Collection<String> paths, final Transcoder<T> transcoder) throws Exception {
+  private <T> List<T> getAsyncThrows(final String pathNameForLogs, final Collection<String> paths, final Transcoder<T> transcoder, final Optional<ZkCache<T>> cache) throws Exception {
     final List<T> objects = Lists.newArrayListWithCapacity(paths.size());
```

Comment:
Outside the scope of this PR but it looks like the `ArrayList` being used here is modified concurrently from multiple threads which seems like it could cause problems